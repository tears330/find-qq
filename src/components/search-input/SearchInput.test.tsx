import { fireEvent, render, screen } from '@testing-library/react';
import { SearchInput } from '.';
describe('SearchInput', () => {
  it('render search input', () => {
    const { baseElement } = render(<SearchInput />);
    expect(screen.getByRole('searchbox')).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  it('controll mode', () => {
    const fn = jest.fn();
    render(<SearchInput value="123123" onChange={fn} />);

    const result = screen.getByRole('searchbox') as HTMLInputElement;
    expect(result.value).toBe('123123');

    fireEvent.change(result, { target: { value: 'test' } });

    expect(fn).toBeCalled();
  });

  it('custom input properties', () => {
    render(<SearchInput type="checkbox" />);
    const result = screen.getByRole('checkbox') as HTMLInputElement;
    expect(result.type).toBe('checkbox');
  });
});
