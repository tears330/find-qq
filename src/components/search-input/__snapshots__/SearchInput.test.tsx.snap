// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput render search input 1`] = `
<body>
  <div>
    <div
      class="com-search"
    >
      <input
        autocomplete="off"
        type="search"
      />
    </div>
  </div>
</body>
`;
